aiohttp_sse/__init__.py

```diff
@@ -142,9 +142,25 @@ async def wait(self) -> None:
         """
         if self._ping_task is None:
             raise RuntimeError("Response is not started")
-        with suppress(asyncio.CancelledError):
+
+        try:
             await self._ping_task
 
+        except asyncio.CancelledError:
+            if (
+                sys.version_info >= (3, 11)  # TODO(PY311): remove this condition
+                and (task := asyncio.current_task())
+                and task.cancelling()
+            ):
+                raise
+            return
+
+        except (ConnectionResetError, RuntimeError):
```

Comment:
I still think the ping task should handle this internally, it's an implementation detail of the task and should work if called elsewhere. I'd like to merge my PR, and then focus on fixing the cancellation behaviour here separately.

Reply:
Okay!
I'll make an issue from your message about cancel swallowing